src/dotnet/Chat.UI.Blazor/Components/NavigateToUnreadOrEnd/NavigateToUnreadOrEnd.razor

```diff
@@ -0,0 +1,103 @@
+@namespace ActualChat.Chat.UI.Blazor.Components
+@inherits ComputedStateComponent<NavigateToUnreadOrEnd.Model>
+@{
+    var m = State.Value;
+    _animator.State = m.IsVisible;
+    if (_animator.MustHideComponent)
+        return;
+    var tooltip = "Navigate to the last message";
+}
+
+<div class="navigate-to-end @_animator.Class">
+    <div class="c-content">
+        @if (m.UnreadCount > 0) {
+            tooltip = "Navigate to the new message";
+            var text = m.UnreadCount.FormatK();
+            <Badge Class="unread-counter" Click="OnClick">@text</Badge>
+        }
+        <ButtonRound Tooltip="@tooltip" Click="OnClick">
+            <i class="icon-arrow-down text-2xl font-thin"></i>
+        </ButtonRound>
+    </div>
+</div>
+
+@code {
+    private ShowHideAnimator _animator = null!;
+    private IMutableState<bool> _overrideToZero = null!;
+
+    [Inject] private Session Session { get; init; } = null!;
+    [Inject] private IChats Chats { get; init; } = null!;
+    [Inject] private ChatUI ChatUI { get; init; } = null!;
+    [Inject] private UIEventHub UIEventHub { get; init; } = null!;
+
+    [Parameter, EditorRequired] public string ChatSid { get; set; } = "";
+    [Parameter, EditorRequired] public bool IsVisible { get; set; }
```

Comment:
Because component client have to assign a property wen component is used.
It's same as in previous implementation. See NavigateToEnd. Due to renamings Git lost track to previous code.